### ISSUE TITLE
Data binding wise controls

### ISSUE THREAD

**Issue by relu91** [nifty-gui/nifty-gui#398]
Hello!
Since I see that after the new layout system the controls were wiped out, I think that we should consider to build them with data binding in mind. it's a must in a modern gui framework to have a advance data binding system . At the title say we could start to provide data binding support at controls level, for two reasons:
1. They are at the higher level on abstraction in new nifty
2. They aren't yet implemented.
3. I don't know if there could be a time when a biding with a specific property of a node it's needed.

What do you think about this? Do you know some binding framework? I saw [Melasse](https://github.com/cchantep/melasse) (pretty easy, I built an example in seconds )

``` java

    public static void main(String[] args) {
        System.out.println("");
        BindingOptionMap opts = new BindingOptionMap();
        Base b = new Base();
        b.setId("123");
        Target t = new Target();
        t.setName("Nifty");

        System.out.println(b.getId());
        System.out.println(t.getName());
        System.out.println();

        melasse.Binder.bind("id", b, "name",t,opts);

        System.out.println(b.getId());
        System.out.println(t.getName());
        System.out.println();

        b.setId("GUI");
        System.out.println(b.getId());
        System.out.println(t.getName());


    }
```

output:

```
123
Nifty

123
123

GUI
GUI
```

**Comment by mkaring** [comment]
I don't know any such framework but creating a simple binding framework is something we can do ourselves. I am not fond of using Melasse because it has a lot of dependencies to Swing and I'd really like to keep any other UI framework out of Nifty-GUI. Unless @void256 has a different opinion.

In any case I agree that we need some nifty (:wink:) data binding. I would go so far as to say that this is something that should be supported by all nodes so we could also apply bindings to the values to layout nodes for example.

**Comment by JurgenCruz** [comment]
Did data binding every was implemented? this would be a very cool addition

**Comment by relu91** [comment]
No, it has never be implemented as far as I know... I stopped developing Nifty Gui a while ago and I did some preliminary tests with Melasse.
